core/src/lib/btime.cc

```diff
@@ -129,26 +177,35 @@ utime_t StrToUtime(const char* str)
   struct tm tm;
   time_t time;
 
+  memset(&tm, 0, sizeof(tm));
+  char* garbage = nullptr;
+
   /*
    * Check for bad argument
    */
   if (!str || *str == 0) { return 0; }
 
-  if (sscanf(str, "%d-%d-%d %d:%d:%d", &tm.tm_year, &tm.tm_mon, &tm.tm_mday,
-             &tm.tm_hour, &tm.tm_min, &tm.tm_sec)
-      != 6) {
+  if ((sscanf(str, "%d-%d-%d %d:%d:%d%s", &tm.tm_year, &tm.tm_mon, &tm.tm_mday,
+              &tm.tm_hour, &tm.tm_min, &tm.tm_sec, &garbage)
+       != 7)
+      || garbage != nullptr) {
```

Comment:
yes exactly. I remarked that if the date is followed with any kind of garbage, it creates an error later on in the process that calls `StrToUtime()`

For example, in the restore command, `StrToUtime(ua->cmd)` is used to check if  the date inside `ua->cmd` is valid. If so, the whole `ua->cmd` is copied into another variable which in turn will be passed for the next step.

In this case, let's take `ua->cmd` is `2020-01-22 12:00:00 cpp`. Due to the nature of `sscanf`, the first part of `ua->cmd` containing the correct date will be scanned and deemed a correct date even though there is a trailing string at the end. In the next step, since `ua->cmd` is deemed correct, all of it is copied to a variable including the trailing string ` cpp`. And that creates errors in other places that are waiting for an actual full date. 

And so now, the trailing string is caught at the level of `StrToUtime()` instead of later stages.